src/devices/device.ts

```diff
@@ -153,6 +234,183 @@ export class Device extends EventEmitter {
+        return this.send(Buffer.from([0x22, this.portId, mode, type]), Consts.BLECharacteristic.LPF2_ALL);
+    }
+
+    private _parseModeInformationResponse (message: Buffer) {
+        const mode = message[4];
+        const debugHeader = `Port ${toHex(this.portId)}, mode ${mode},`;
+        const type = message[5];
+        switch (type) {
+            case 0x00: { // Mode Name
+                const name = message.slice(6, message.length).toString().replace(/\0/g, '');
+                modeInfoDebug(`${debugHeader} name ${name}`);
+                if (this.autoParse) {
+                    this._modes[mode].name=name;
+                }
+                break;
+            }
+            case 0x01: { // RAW Range
+                const min = message.readFloatLE(6);
+                const max = message.readFloatLE(10);
+                modeInfoDebug(`${debugHeader} RAW min ${min}, max ${max}`);
+                if (this.autoParse) {
+                    this._modes[mode].raw.min=min
+                    this._modes[mode].raw.max=max;
+                }
+                break;
+            }
+            case 0x02: { // PCT Range
+                const min = message.readFloatLE(6);
+                const max = message.readFloatLE(10);
+                modeInfoDebug(`${debugHeader} PCT min ${min}, max ${max}`);
+                if (this.autoParse) {
+                    this._modes[mode].pct.min=min;
+                    this._modes[mode].pct.max=max;
+                }
+                break;
+            }
+            case 0x03: {// SI Range
+                const min = message.readFloatLE(6);
+                const max = message.readFloatLE(10);
+                modeInfoDebug(`${debugHeader} SI min ${min}, max ${max}`);
+                if (this.autoParse) {
+                    this._modes[mode].si.min=min;
+                    this._modes[mode].si.max=max;
+                }
+                break;
+            }
+            case 0x04: {// SI Symbol
+                const symbol = message.slice(6, message.length).toString().replace(/\0/g, '');
+                modeInfoDebug(`${debugHeader} SI symbol ${symbol}`);
+                if (this.autoParse) {
+                    this._modes[mode].si.symbol=symbol;
+                }
+                break;
+            }
+            case 0x80: {// Value Format
+                const numValues = message[6];
+                const dataType = message[7];
+                const totalFigures = message[8];
+                const decimals = message[9];
+                modeInfoDebug(`${debugHeader} Value ${numValues} x ${dataType}, Decimal format ${totalFigures}.${decimals}`);
+                if (this.autoParse) {
+                this._modes[mode].values.count=numValues;
+                this._modes[mode].values.type=dataType;
+
+                    if (mode === this._modeCount - 1) {
+                        this._init();
+                    }
```

Comment:
Call `init` when we are done with mode information messages (and `autoParse` enabled)